lib/board.rb

```diff
@@ -10,9 +10,40 @@ def initialize
          @cells[index] = Cell.new(index)
        end
      end
+
+		 ###########
+		 @all_cells = []
+		 @all_cells = @cells.keys
+		 @hor_arr = []
```

Comment:
We also already have this inside of the valid_placement? method, so we don't need it as an instance variable